osquery/logger/plugins/http_logger.cpp

```diff
@@ -0,0 +1,104 @@
+/*
+ *  Copyright (c) 2014, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. An additional grant 
+ *  of patent rights can be found in the PATENTS file in the same directory.
+ *
+ */
+
+#include <exception>
+#include <mutex>
+
+#include <boost/property_tree/ptree.hpp>
+#include <boost/property_tree/json_parser.hpp>
+
+#include <osquery/filesystem.h>
+#include <osquery/flags.h>
+#include <osquery/logger.h>
+#include <osquery/registry.h>
+#include <osquery/database.h>
+
+#include "osquery/dispatcher/dispatcher.h"
+
+namespace osquery {
+
+const std::string kLoggerPluginKey = "http_logger";
+class LogForwarder;
+
+FLAG(string,
+     logger_uri,
+     "",
+     "Endpoint path for ERROR/WARN/INFO and results logging");
+
+class HTTPLoggerPlugin : public LoggerPlugin {
+ public:
+  Status setUp();
+  Status logString(const std::string& s);
+  Status init(const std::string& name, const std::vector<StatusLogLine>& log);
+  Status logStatus(const std::vector<StatusLogLine>& log);
+};
+
+REGISTER(HTTPLoggerPlugin, "logger", "http");
+
+Status HTTPLoggerPlugin::setUp() {
+  return Status(0, "OK");
+}
+
+Status HTTPLoggerPlugin::logString(const std::string& s) {
+  auto stat = DBHandle::getInstance()->Put(kEvents, kLoggerPluginKey, s);
+  return stat;
+}
+
+Status HTTPLoggerPlugin::logStatus(
+    const std::vector<StatusLogLine>& log) {
+  for (const auto& item : log) {
+    boost::property_tree::ptree new_log;
+    new_log.put("action", "status_log");
+    new_log.put("severity", (google::LogSeverity)item.severity);
+    new_log.put("content", item.filename + " " + std::to_string(item.line) + " " + item.message);
+    std::stringstream ss;
+    write_json(ss, new_log);
+    auto stat = DBHandle::getInstance()->Put(kEvents, kLoggerPluginKey, ss.str());
+    if(!stat.ok()){
+      return stat;
+    }
+  }
+  return Status(0, "OK");
+}
+
+Status HTTPLoggerPlugin::init(const std::string& name,
+                                    const std::vector<StatusLogLine>& log) {
+  // Stop the internal glog facilities.
+  google::ShutdownGoogleLogging();
+
+  PluginResponse resp;
+  Status s = Registry::call("enrollment", "get_key", {{"enroll", "0"}}, resp);
+
+  // If the call was successful and returned an enrollment key
+  if (!s.ok() || resp.size() == 0 || resp[0]["key"].length() == 0) {
+    // We could not get an enrollment key, so we can send logs, best we can do
+    // is log to stderr
+    FLAGS_logtostderr = true;
+  }
+  
+  // Restart the glog facilities using the name init was provided.
+  google::InitGoogleLogging(name.c_str());
+  Status stat = logStatus(log);
+  if(!stat.ok()){
+    return stat;
+  }
+  return Status(0, "OK");
+}
+
+class LogForwarder : public InternalRunnable{
```

Comment:
I feel like this runnable class should be somewhere else, maybe in `osquery/logger/log_forwarder.cpp` and `include/osquery/logger/log_forwarder.h`?